zokrates_core/src/semantics.rs

```diff
@@ -684,7 +794,7 @@ impl<'ast, T: Field> Checker<'ast, T> {
                                         }),
                                         ..t
                                     }),
-                                    _ => unreachable!()
+                                    _ => t // type alias
```

Comment:
If I understand correctly, the `DeclarationType::Struct` case in this match can happen for a struct declaration, or for an alias declaration which points to a struct. In the latter case, I don't understand why the Struct type location should be modified?

Reply:
Maybe this should also match `alias_generics` as if I understand correctly, this is what determines whether we have a struct declaration or an alias declaration. A similar match happens further down over both.